File: Number Patterns/number-square/zigZag/Square/number/ziz-zag-number-square.py
numRows = 5
numToPrint = 1;

for rowVal in range(1, numRows + 1):
    for colVal in range(1, numRows + 1):
        if rowVal % 2 == 1:
            # heer, I am only using this "" because, sep parameter is already giving us one space as sep=" "
            # and also using pyhton based number formatting to format the output. 
            print("%3d" % (numToPrint), "", end="")
            numToPrint += 1 
        else:
            # heer, I am only using this "" because, sep parameter is already giving us one space as sep=" "
            # and also using pyhton based number formatting to format the output. 
            print("%3d" % (numToPrint), "", end="")
            numToPrint -= 1
    # this is how we can use ternary operator in python
    numToPrint += numRows + (-1 if rowVal % 2 == 1 else 1);
    print()   # for new line

File: Number Patterns/number-square/zigZag/Square/number/ziz-zag-number-square.c
#include<stdio.h>

void main() {
    int numRows = 5;
    int rowVal;
    int colVal;
    int numToPrint = 1;
    for(rowVal = 1; rowVal <= numRows; rowVal++) {
        for(colVal = 1; colVal <= numRows; colVal++) {
            if(rowVal % 2 == 1) {
                // Logic for Odd row
                printf("%3d", numToPrint);
                numToPrint++;
            } else {
                // Logic for Even row
                printf("%3d", numToPrint);
                numToPrint--;
            }
        }
        // for true condition, it will returns -1 else return 1.
        numToPrint += numRows + ((rowVal % 2 == 1) ? -1 : 1);
        printf("\n");
    }
}